src/core.ts

```diff
@@ -124,6 +126,24 @@ export function instrument(eventState: IInstrumentationEventState) {
   }
 }
 
+export function onWorkerMessage(evt: MessageEvent) {
+  let message = JSON.parse(evt.data) as IWorkerMessage;
+  switch (message.type) {
+    case WorkerMessageType.Upload:
+      let uploadMsg = message as IUploadMessage;
+      let onSuccess = (status: number) => { mapProperties(droppedPayloads, uploadDroppedPayloadsMappingFunction, true); };
+      let onFailure = (status: number) => { onFirstSendDeliveryFailure(status, uploadMsg.rawData, uploadMsg.compressedData); };
+      let compressedKb = Math.ceil(uploadMsg.compressedData.length / 1024.0);
+      let rawKb = Math.ceil(uploadMsg.rawData.length / 1024.0);
+      let envelope = JSON.parse(uploadMsg.rawData).envelope as IEnvelope;
```

Comment:
nit: we are doing these calculations even if we aren't going to use them - can we inline them in the debug call to make sure we don't do any unnecessary work?

Reply:
Good catch

---
In reply to: [136671385](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136671385) [](ancestors = 136671385)